### DISCUSSION TITLE
Custom App

### DISCUSSION THREAD

**Discussion by martinRenou** [jupyterlite/jupyterlite#772] [Q&A] [answered]
I was looking into making an app from the outside from JupyterLite (maybe as an Addon). Is there a way to do this?

**Comment by jtpio** [comment, marked as answer]
There are a few mentions to this in several issues nothing in the works yet as far as I know.

Just like for extensions, the idea would be to support federated applications probably via an addon yes. Maybe this requires a new entry under `package.json#jupyterlite`. The federated app assets could then be copied to the `{output-dir}/app` folder at build time.

**Reply by martinRenou** [reply]
Yes makes sense. Thanks for your answer.

Would you be open to add a new Voilite app in the JupyterLite repo, in the meantime?

**Reply by jtpio** [reply]
Since Voila is an official Jupyter sub-project it would make sense to have it here I think yes.

That way it will also benefit from the single webpack build to reduce the overall footprint. Site deployers can always remove it at build time if they want to (https://jupyterlite.readthedocs.io/en/latest/howto/configure/advanced/optimizations.html#removing-applications).

**Reply by martinRenou** [reply]
Great! I think I'll do that.

At first I wanted to only have a Voilite nbconvert exporter. But it seems that benefiting from the whole federated labextension and Addons setup from JupyterLite would be beneficial.
